gateway-ha/src/test/java/io/trino/gateway/ha/clustermonitor/TestClusterMetricsStatsExporter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.clustermonitor;
+
+import io.airlift.units.Duration;
+import io.trino.gateway.ha.config.MonitorConfiguration;
+import io.trino.gateway.ha.config.ProxyBackendConfiguration;
+import io.trino.gateway.ha.router.GatewayBackendManager;
+import org.junit.jupiter.api.Test;
+import org.weakref.jmx.MBeanExporter;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+final class TestClusterMetricsStatsExporter
+{
+    @Test
+    void testMetricsRegistrationForNewCluster()
+            throws InterruptedException
+    {
+        try (ClusterMetricsStatsExporter statsExporter = createStatsExporter()) {
+            String clusterName1 = "test-cluster1";
+            ProxyBackendConfiguration cluster1 = createTestCluster(clusterName1);
+            String clusterName2 = "test-cluster2";
+            ProxyBackendConfiguration cluster2 = createTestCluster(clusterName2);
+            when(statsExporter.getGatewayBackendManager().getAllBackends())
+                    .thenReturn(List.of(cluster1)) // First return with 1 cluster
+                    .thenReturn(List.of(cluster1, cluster2)); // Then return with 2 clusters to simulate addition
+
+            statsExporter.start();
+            Thread.sleep(2000);
```

Comment:
Then we can use the exposed method instead for testing.
```suggestion
            statsExporter.updateClustersMetricRegistry();
```

Reply:
Since the updating of the clusters metrics registry is done on a periodic basis, I think it makes sense to just use sleep rather than explicitly call the update method to test the refresh logic. The statsExporter we create for this test has a set refresh period of 1 second, so sleeping 2 seconds would simulate going through at least one refresh cycle.